src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -457,9 +752,9 @@ def can_loop_be_parallelised(self, loop, loop_variable=None,
         self._clear_messages()
 
         if not isinstance(loop, Loop):
-            raise TypeError("can_loop_be_parallelised: node must be an "
-                            "instance of class Loop but got '{0}'".
-                            format(type(loop).__name__))
+            raise TypeError(f"can_loop_be_parallelised: node must be an "
+                            f"instance of class Loop but got "
+                            f"'{type(loop).__name__}'")
         if not loop_variable:
```

Comment:
This is not strictly touched in this PR but I had this question before:

is the `loop_variable` argument not always `loop.variable`? and the `var_accesses` argument, not always the result of `loop.reference_accesses(var_accesses)`?.  In what context do we need them to be provided as arguments?

I guess for "var_accesses" it could be a performance benefit to not traverse again the tree if we already have it, but I am still not convinced since any mutation of the tree could  produce a different var_access. And the user must know that is must not be re-used if anything has changed. So I think it would be safer to not provide this optional argument. But maybe there is some place where this is needed?

Reply:
Done.